File: python3/3115.py
class Solution:
    def maximumPrimeDifference(self, nums: List[int]) -> int:
        def prime(n):
            if n == 1:
                return False
            for i in range(2, int(sqrt(n)) + 1):
                if n % i == 0:
                    return False
            return True

        s = 0
        e = len(nums) - 1
        while s <= e:
            if prime(nums[s]):
                break
            s += 1
        while s <= e:
            if prime(nums[e]):
                return e - s
            e -= 1
        return 0

File: cpp/3115.cpp
class Solution {
private:
    bool prime(int n) {
        if (n == 1)
            return false;
        int s = 2, e = (int)sqrt(n);
        while (s <= e)
            if (n % s++ == 0)
                return false;
        return true;
    }

public:
    int maximumPrimeDifference(vector<int>& nums) {
        int s = 0, e = nums.size() - 1;
        while (s <= e)
            if (prime(nums[s++]))
                break;
        s--;
        while (s <= e)
            if (prime(nums[e--]))
                return e - s + 1;
        return e - s + 1;
    }
};
